### PR TITLE
Added support for different types of http clients.

### PR DESCRIPTION
**Pull request by marcoaltiericoupa**
For the integration between CLMA and C.E. we are using okhttp.

### CODE REVIEW

**Comment by marcoaltiericoupa** [comment]
@coupa-toddsavage, I added your classes with a minor change.

**Comment by marcoaltiericoupa** [comment]
@johnwu96822 This change has been necessary because our java client is based on "Retrofit2" and OkHttp. We couldn't return an HttpResponse. In this P.R. we are using a new different interface that can wrap the actual Response.
We could have rewritten the existing method using the new interface but we preferred to reduce the changes.

File: src/main/java/com/coupa/sand/Client.java [hunk 1]
```diff
@@ -173,6 +174,100 @@ public HttpResponse request(String keyForCaching,
...
+            String[] scopes,
+            Function<String, GenericResponse<R>> requestFunction) {
+
+        return genericRequest(keyForCaching, scopes, DEFAULT_RETRY_COUNT, requestFunction);
+    }
+
+    /**
+     * Makes a Client request to a Service by applying a token to the requestFunction parameter.
+     * Will fetch new tokens and perform retries (at least one) if the response is 401.
+     * Will use the default retry count;
```
Comment (djarlx) [inline src/main/java/com/coupa/sand/Client.java]: This function is not using the default retry count.

Comment (marcoaltiericoupa) [reply src/main/java/com/coupa/sand/Client.java]: @djarlx The default retry count is passed to the clientRequestRetryCount() method. Unless I am missing something, @coupa-toddsavage as used the same logic that was implemented in the original request() method.

Comment (djarlx) [reply src/main/java/com/coupa/sand/Client.java]: The comment is for the function below that takes a retries parameter, which the function above sets to DEFAULT_RETRY_COUNT. So the function above uses default, this function you can set your own retries count.

Comment (marcoaltiericoupa) [reply src/main/java/com/coupa/sand/Client.java]: I fixed the javadoc. Thanks.

**Comment by marcoaltiericoupa** [comment]
@djarlx @johnwu96822 Could you confirm that I can now merge the changes?

**Pull request merged**